rcl/include/rcl/node.h

```diff
@@ -236,8 +234,8 @@ rcl_node_get_default_options(void);
  * <i>[1] if `atomic_is_lock_free()` returns true for `atomic_uint_least64_t`</i>
  *
  * \param[in] node rcl_node_t to be validated
- * \param[in] allocator external allocator pointer used for allocations.
- * \return `true` if the node is valid, otherwise `false`.
+ * \param[in] allocator external allocator pointer used for allocations
+ * \return `true` if the node and allocator are valid, otherwise `false`.
```

Comment:
Not sure how to capture that passing a `NULL` allocator is valid too, maybe:
```
\param[in] allocator external allocator pointer used for allocations, must be a valid allocator or NULL
  a NULL allocator will result in using the default allocator
```

Reply:
how about ca141f2d01d7741f2afbdfd72a8d406cd40feb32

Reply:
If I need to put more extensive qualifications on a parameter, I do so in prose in the body of the doc block, and try to keep the parameter description succinct.

I would do something like this:

```
 * The allocator needs to either be a valid allocator or NULL, in which case
 * the default allocator will be used.
 * The allocator is used when allocation is needed (like for an error message)
 * but the node's allocator is invalid or cannot be checked.
...
 * \param[in] allocator a valid allocator or NULL
```

Reply:
in cfde9b3a34df97ae2c595cae88717ad160483bab I mostly copied that @wjwwood but FWIU the node's allocator is never used in this function

Reply:
Oh, well I just assumed it would try to use that if it were valid, but I guess that's the last case checked.